master/internal/stream/system.go

```diff
@@ -467,6 +471,22 @@ func prepareMessageWithCache(
 	return *cache
 }
 
+func prepareWebsocket(msg interface{}) interface{} {
```

Comment:
I'd like to request that you move this function to live next to the one place where it is used.  I think code is a lot easier to understand if the amount of jumping needed to read it is minimized.